app/services/tagging/tagging_update_form.rb

```diff
@@ -40,6 +41,24 @@ def related_item_errors
       @related_item_errors ||= {}
     end
 
+    def title_for_related_link(base_path)
+      link = links.ordered_related_items.find { |related_item| related_item.fetch("base_path") == base_path }
+
+      if link.nil?
+        # links is populated from the existing links. When we submit a form and
+        # it fails, we may have additional related links we want to know the
+        # titles of.
+        # In practice this shouldn't actually be visible most of the time,
+        # because with javascript enabled, the user can't enter invalid data,
+        # and without it we don't display the titles. It is still possible to
+        # see errors due to tagging conflicts though.
+        # TODO: attempt to fetch missing links on demand.
```

Comment:
I'm okay with doing it as-is. Not sure if @alecgibson's card will affect this particular code (but keep it in mind).